test/parallel/test-stream-map.js

```diff
@@ -158,6 +158,21 @@ const { setTimeout } = require('timers/promises');
   });
 }
 
+{
+  // Allow Infinite concurrency
+  const stream = Readable.from([1, 2]).map(async (item, { signal }) => {
+    await setTimeout(10 - item, { signal });
```

Comment:
I am not sure this asserts concurrency works

Reply:
The test meant to make sure that `concurrency: Infinity` does not throw an error, but you are right, it does not test that concurrency works, fixed it

Reply:
Why do we need a timer here? Can't this just be something like:

```js
let count = 0; // or t.mock
Readable.from(Array(1000).fill()).map(() => {
  count++;
  return new Promise(() => {});
}, { concurrency: Infinity }).toArray().catch(common.mustNotCall());

await setImmediate(); // from timers/promises
strictEqual(count, 1000);
```